daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1062,11 +1062,13 @@ private[lf] object SBuiltin {
       val coid = getSContractId(args, 0)
       onLedger.cachedContracts.get(coid) match {
         case Some(cached) =>
-          onLedger.ptx.consumedBy
-            .get(coid)
-            .foreach(nid =>
+          onLedger.ptx.consumedByOrInactive(coid) match {
+            case Some(Left(nid)) =>
               throw SErrorDamlException(IE.ContractNotActive(coid, cached.templateId, nid))
-            )
+            case Some(Right(())) =>
+              throw SErrorDamlException(IE.ContractNotFound(coid))
```

Comment:
Let’s leave it for this PR but I think this maybe should throw ContractNotActive and we should make the node id optional. ContractNotFound is if the contract is not known to the ledger or speedy. Here we know about the contract but the create was rolled back.